### DISCUSSION TITLE
Github API for updating a file content is not working for files under .github folders. Could you please suggest solution for that?

### DISCUSSION THREAD

**Discussion by dhananjayagupta** [community/community#24685] [Apps, API and Webhooks]
<p>Hi All,</p>
<p>I facing a strange behavior here in case of updating a file content using github API.</p>
<p>curl API command: <a href="https://api.github.com/repos/" rel="noopener nofollow ugc">https://api.github.com/repos/</a>&lt;internal_org&gt;/&lt;repo_name&gt;/contents/.github/workflows/TEST_pipeline.yml</p>
<p>data = {<br>
“message”: “Updating via api”,<br>
“content”: “base64 encoded data”,<br>
“sha”: “sha for the file”<br>
}</p>
<p>Problem: Except .github folder the file update is happening everywhere via above API call. I am receiving below error if I am doing the file updation for files under .github folder .</p>
<p>{“message”:“Not Found”,“documentation_url”:“<a href="https://docs.github.com/rest/reference/repos#create-or-update-file-contents" class="inline-onebox" rel="noopener nofollow ugc">Repositories - GitHub Docs</a>”}</p>
<p>Error Code : 404</p>
<p>Can any one suggest the solution here.</p>
<p>Thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The <strong>repo</strong> scope isn’t sufficient for updating workflow files, you need to have the <strong>workflow</strong> scope, too. See <a href="https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes" class="inline-onebox">Scopes for OAuth Apps - GitHub Docs</a>.</p>
